### PR TITLE
[FIXED] Allow leafnode authorization to support nkeys.

### PR DESCRIPTION
**Pull request by derekcollison**
Resolves: #3196 

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/opts.go [hunk 1]
```diff
@@ -2351,6 +2355,12 @@ func parseLeafAuthorization(v interface{}, errors *[]error, warnings *[]error) (
 			auth.user = mv.(string)
 		case "pass", "password":
 			auth.pass = mv.(string)
+		case "nkey":
+			nk := mv.(string)
```
Comment (neilalexander) [inline server/opts.go:2359]: Is this guaranteed to be `string` or do we want an `nk, ok :=` to avoid a possible panic?

File: server/opts.go [hunk 2]
```diff
@@ -2496,7 +2506,24 @@ func parseRemoteLeafNodes(v interface{}, errors *[]error, warnings *[]error) ([]
...
 					continue
 				}
+				// Can't have both creds and nkey
+				if remote.Nkey != _EMPTY_ {
+					*errors = append(*errors, &configErr{tk, "Remote leafnode can not have both creds and nkey defined"})
+					continue
+				}
 				remote.Credentials = p
+			case "nkey", "seed":
+				nk := v.(string)
```
Comment (neilalexander) [inline server/opts.go:2516]: Same here

**Review by neilalexander — APPROVED** [review]
LGTM

File: server/opts.go (hunk 1 above)
Comment (derekcollison) [reply server/opts.go:2359]: Not guaranteed but like above do not test for failed cast, so I think its ok.

**Pull request merged**